pkg/cmdimpl/add.go

```diff
@@ -349,134 +270,276 @@ func Add(args []string, allParams AddParamSet, deps *AddDependencies) error {
+	}
+	applicationGoat, err := generateApplicationGoat(userSourceName)
+	if err != nil {
+		return wrapError(err, fmt.Sprintf("could not create GitOps automation for '%s'", params.Name))
+	}
+	if err := writeAppYaml(gitClient, appYaml, ".wego"); err != nil {
+		return err
+	}
+	if err := writeGoats(gitClient, ".wego", applicationGoat); err != nil {
+		return err
+	}
+	return commitAndPush(ctx, gitClient)
+}
 
-	gitClient := deps.GitClient
+func addAppWithConfigInExternalRepo(ctx context.Context, gitClient git.Git) error {
+	// Source covers entire GOAT repo
+	goatSourceName := getGoatRepoName()
+	tmpDir, err := cloneToTempDir(ctx, gitClient) // leaves gitClient referencing cloned repo
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
 
-	wegoRepoURL := fmt.Sprintf("ssh://git@github.com/%s/%s.git", owner, wegoRepoName)
-	fmt.Printf("Cloning %s...\n", wegoRepoURL)
-	if !params.DryRun {
-		if _, err := gitClient.Clone(ctx, wegoRepoDir, wegoRepoURL, params.Branch); err != nil {
-			return wrapError(err, fmt.Sprintf("could not clone repository: %s", wegoRepoURL))
-		}
+	goatRepoSource, err := generateSource(goatSourceName, params.AppConfigUrl, SourceTypeGit)
+	if err != nil {
+		return err
+	}
+	goatTargetKustomize, err := generateTargetKustomize(goatSourceName, ".")
+	if err != nil {
+		return err
+	}
+	goatAppKustomize, err := generateAppKustomize(goatSourceName, ".")
+	if err != nil {
+		return err
+	}
+	if err := applyToCluster(goatRepoSource, goatTargetKustomize, goatAppKustomize); err != nil {
+		return err
+	}
+	// Source covers entire user repo
+	userSourceName := getUserRepoName()
+	userRepoSource, err := generateSource(userSourceName, params.Url, SourceType(params.SourceType))
+	if err != nil {
+		return err
+	}
+	userTargetKustomize, err := generateTargetKustomize(userSourceName, ".")
+	if err != nil {
+		return err
+	}
+	userAppKustomize, err := generateAppKustomize(userSourceName, ".")
+	if err != nil {
+		return err
 	}
+	appYaml, err := generateAppYaml()
+	if err != nil {
+		return wrapError(err, fmt.Sprintf("could not create app.yaml for '%s'", params.Name))
+	}
+	applicationGoat, err := generateApplicationGoat(userSourceName)
+	if err != nil {
+		return wrapError(err, fmt.Sprintf("could not create GitOps automation for '%s'", params.Name))
+	}
+	if err := writeAppYaml(gitClient, appYaml, "."); err != nil {
+		return err
+	}
+	if err := writeGoats(gitClient, "", userRepoSource, userTargetKustomize, userAppKustomize, applicationGoat); err != nil {
+		return err
+	}
+	return commitAndPush(ctx, gitClient)
+}
 
-	fmt.Fprintf(shims.Stdout(), "Applying wego platform resources...\n")
-	if !params.DryRun {
-		// Install Source and Kustomize controllers, and CRD for application (may already be present)
-		wegoSource, err := generateWegoSourceManifest()
-		if err != nil {
-			return wrapError(err, "could not generate wego source manifest")
-		}
+func generateSource(repoName, repoUrl string, sourceType SourceType) ([]byte, error) {
```

Comment:
We aren't using sourceType in the function.  (I suspect it's because we have a different PR that will add the helm support)